airflow/providers/ssh/hooks/ssh.py

```diff
@@ -173,6 +182,14 @@ def __init__(
                 if "conn_timeout" in extra_options and self.conn_timeout is None:
                     self.conn_timeout = int(extra_options["conn_timeout"])
 
+                if self.cmd_timeout is None:
+                    if "cmd_timeout" in extra_options:
+                        self.cmd_timeout = (
+                            int(extra_options["cmd_timeout"]) if extra_options["cmd_timeout"] else None
+                        )
+                    else:
+                        self.cmd_timeout = CMD_TIMEOUT
+
```

Comment:
This seems like some inconsistent between direct hook parameters and Connection parameters.
You can turn off cmd timeout only by connection, but not by hook parameters, it always fallback to `CMD_TIMEOUT`

Usual direct hook parameters override connection

Reply:
Just for a record, we have a specific sentinel type and value, which use when None as default value for argument is not an option.

https://github.com/apache/airflow/blob/d67ac5932dabbf06ae733fc57b48491a8029b8c2/airflow/utils/types.py#L28-L44


So potentially we could use it here, e.g.:

If hook argument cmd_timeout is NOTSET and Connection extra argument set than use from connection
If hook argument cmd_timeout is NOTSET and Connection extra argument **not set** than use CMD_TIMEOUT 
If hook argument set to any value rather than NOTSET than use this value and ignore value from connection


And of course we need to tests all combination in unittests, just for sure that we do not break anything as well as we correctly process every possible case

Reply:
Hi, @Taragolis  ! Good point about inconsistency between direct hook parameters and connection parameters - I changed  it to `negative value means no timeout` (and now it's possible to set it both in the hook and in the connection).

However I'm not sure I agree with NOTSET approach. It will (somewhat) bring back the issue with inconsistency between hook/connection parameters when you try to turn off cmd_timeout. Because in this case:

- 	To pass "no timeout" via hook parameters you should pass `None` as a hook argument.
- 	To pass "no timeout" via connection parameters you should specify `null` in connection extra: `{"cmd_timeout": null}`. (It's possible to use empty string or negative values instead of null here, but I feel like it'll be even more inconsistent with hook parameters).

I fill like specifying a parameter in connection extra and setting it to null is a bit counterintuitive. If you think it's ok, I'll change it to NOTSET approach, but I feel like negative values both in hook and connection for `no timeout` would be a little more clear.

The unittests are already present at https://github.com/apache/airflow/pull/29347/files#diff-8a4026fba1b53549c99ae768b6308aefddee93e54f3c8105aa454a15ee7cc075R824 .

Reply:
> To pass "no timeout" via hook parameters you should pass None as a hook argument.
To pass "no timeout" via connection parameters you should specify null in connection extra: {"cmd_timeout": null}. (It's possible to use empty string or negative values instead of null here, but I feel like it'll be even more inconsistent with hook parameters).

I can't see any inconsistent here if you want so set no timeout, there is no problem to set it to `None` or json `null` explicitly, rather than use negative values which converted to `None` anyway. 

NOTSET use as sentinel in situation when None value itself it is a legit value but for some reason we can't provide `None` as default value, and in this case it is 100% legit because  [paramico use `None` as no timeout](https://docs.paramiko.org/en/stable/api/client.html#paramiko.client.SSHClient.exec_command).

Potentially this discussion about `spaces vs tabs`

Reply:
As per suggestion, changed to `None/null means no timeout`.